### PR TITLE
Add jsdoc comments to all of the functions

### PR DESCRIPTION
**Pull request by chee**
some of these are quite extravagant

### CODE REVIEW

File: lib/babel.js [hunk 1]
```diff
@@ -35,13 +62,26 @@ export function createConfiguration ({
...
 			error ? reject(error) : resolve(result.code)
 		})
 	})
 
+/**
+ * Build a given filepath to a target path with our babel config
+ * @param {string} sourceFile the path to the source code
+ * @param {string} targetFile the path to output on
+ * @param {ConfigurationOptions} options Options to pass to createConfiguration
+ * @returns {Promise.<void>} resolves when done, rejects if couldn't
```
Comment (rowanmanning) [inline lib/babel.js]: Should this be:

```suggestion
 * @returns {Promise<void>} resolves when done, rejects if couldn't
```

(basing this assumption on the previous method using `{Promise<string>}`)

File: lib/babel.js [hunk 2]
```diff
@@ -54,14 +94,22 @@ async function buildFile (sourceFile, targetFile, options = {}) {
 	await fs.writeFile(targetFile, targetCode, "utf-8")
 }
 
-let javascriptExtensions = [".js"]
+let JAVASCRIPT_FILE_EXTENSIONS = [".js"]
```
Comment (rowanmanning) [inline lib/babel.js]: Should this be a `const`?

```suggestion
const JAVASCRIPT_FILE_EXTENSIONS = [".js"]
```

File: lib/npm.js [hunk 3]
```diff
@@ -30,11 +78,23 @@ export let mergeManifests = (existing, generated) => {
 	return result
 }
 
+/**
+ * Create the component's npm name from its bower name
+ * @param {string} componentName the name of the component
+ * @param {string} npmOrganisation the organisation to use (default: `financial-times`)
```
Comment (rowanmanning) [inline lib/npm.js]: Shouldn't stop merging, but we could use an optional param with a default here. Thoughts?

```suggestion
 * @param {string} [npmOrganisation="financial-times"] the organisation to use
```

File: lib/babel.js (hunk 1 above)
Comment (chee) [reply lib/babel.js]: oh, i think it's meant to be `Promise.<string>` in the other place

File: lib/babel.js [hunk 4]
```diff
@@ -35,13 +62,26 @@ export function createConfiguration ({
 	}
 }
 
+/**
+ * Run babel transform, return a promise that would resolve with the code
+ * @param {string} code the source code
+ * @param {ConfigurationOptions} options options to pass to createConfiguration
+ * @returns {Promise<string>} the compiled code
```
Comment (chee) [inline lib/babel.js]:
```suggestion
 * @returns {Promise.<string>} the compiled code
```

**Pull request merged**